actionpack/lib/action_controller/metal/request_forgery_protection.rb

```diff
@@ -253,6 +300,57 @@ def handle_unverified_request
       end
     end
 
+    module TokenStorageMethods
+      class Base
```

Comment:
Nesting in `TokenStorageMethods` is quite a mouthful. Namespace are for avoiding name conflicts, not taxonomy, I think we can go with much simpler:

  - `AbstractStore`
  - `CookieStore`
  - `SessionStore`